taichi/ui/backends/vulkan/scene.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+#include "taichi/ui/common/scene_base.h"
+
+TI_UI_NAMESPACE_BEGIN
+
+namespace vulkan {
+
+class Scene final : public SceneBase {
+ public:
+  friend class Canvas;
+  friend class Particles;
+  friend class Mesh;
+
+ private:
+  struct SceneUniformBuffer {
+    alignas(16) glm::vec3 camera_pos;
+    alignas(16) glm::mat4 view;
+    alignas(16) glm::mat4 projection;
+    // vec4 instead of vec3 because of alignment in vulkan uniform buffer
+    glm::vec4 point_light_positions[kMaxPointLights];
+    glm::vec4 point_light_colors[kMaxPointLights];
```

Comment:
use
`
struct Light {
vec4 pos;
vec4 color;
}
`

And then place this array in the end of the UBO to support unlimited size